core/src/main/java/org/mule/runtime/core/api/MuleMessage.java

```diff
@@ -113,7 +114,13 @@ static MuleMessage of(Object payload)
         Builder payload(Object payload);
 
         @Override
+        CollectionBuilder collectionPayload(Iterator payload, Class<?> itemType);
```

Comment:
After thinking about this naming I prefer streamPayload.  Why? Because we will eventually use this not just for sequences of messages (which are also streams, we are used to thinking about them as a collection) but also for streams of ByteBuffers as we move from blocking InputStream streaming to non-blocking streaming.  And given this collectionPayload(Iterator<ByteBuffer>, ByteBuffer.class) or collectionPayload(Publisher<ByteBuffer>, ByteBuffer.class) don't work for me, when these are quite clearly streams and not finite.